### PR TITLE
Migrate ManagePremiums to API4

### PR DESCRIPTION
**Pull request by braders**
Overview
----------------------------------------
Migrate ManagePremiums to API4

Before
----------------------------------------
The primary motivation for this change was to avoid PHP warnings due to DAO objects not assigning NULL values as properties:

Screenshot 2022-11-05 at 22 07 45

More generally moving to API v4 doesn't hurt.

After
----------------------------------------
The warnings are gone.

### CODE REVIEW

File: CRM/Contribute/Page/ManagePremiums.php [hunk 1]
```diff
@@ -107,36 +107,35 @@ public function run() {
...
-    $dao->find();
-
-    while ($dao->fetch()) {
-      $premiums[$dao->id] = [];
-      CRM_Core_DAO::storeValues($dao, $premiums[$dao->id]);
-      // form all action links
-      $action = array_sum(array_keys($this->links()));
-
-      if ($dao->is_active) {
+    $premiums = \Civi\Api4\Product::get()
```
Comment (braders) [inline CRM/Contribute/Page/ManagePremiums.php:110]: I think we want permissions to be checked here, but I can get slightly confused at times in what context we do or don't want to pass in `FALSE` so this wants checking on review.

**Comment by colemanw** [comment]
2 thoughts:

1. This screen is a candidate for reimplementation in SearchKit. See https://lab.civicrm.org/dev/core/-/issues/3912
2. Please take care that html entities are still being escaped properly. The big difference between fetching records from the DAO vs the API is that the former usually returns `<` and `>` characters html-escaped, the latter does not.

**Comment by colemanw** [comment]
I'm closing this for now as there is unresolved discussion. @braders please feel free to reopen when you're able to reply to those. Thanks!

**Comment by eileenmcnaughton** [comment]
Just noting that the second item of the ones mentioned is the more important one. I think the possibility of SearckKit is non-blocking and it is good to get rid of old code regardless 

However, the tpl file one is a bit more of a thing

Basically if html is stored in the database escaped then the apiv4 unescapes it but the DAO fetch does not - which means assigning directly from a DAO retrieval allows us to be sloppier with our in-template escaping than using the api.

So, the steps to get this merged are
1) identify how the field output is exposed to the tpl layer
2) check that the relevant places have escaping (eg. `|escape`)
3) open a new pr from the branch, referencing this one, with commenst about how that was covered off - note opening a new PR puts it back on the front page where it gets seem

**Pull request closed without merging**